MdeModulePkg/Universal/PCD/Pei/Service.c

```diff
@@ -453,6 +461,14 @@ BuildPcdDatabase (
 
   CallbackFnTable = BuildGuidHob (&gEfiCallerIdGuid, SizeOfCallbackFnTable);
 
+  // MU_CHANGE [BEGIN] - CodeQL change
+  if (CallbackFnTable == NULL) {
+    DEBUG ((DEBUG_ERROR, "Failed build CallbackFnTable guid hob.\n"));

```

Comment:
Some of these return paths are going to leave GUID HOBs (like `gPcdDataBaseHobGuid`) around. Something to think about potentially handling in case the caller sees the error and tries again. It's unlikely they'll be able to do anything though.

Reply:
Ties into the developing theme of the PR - how to handle fatal errors in certain paths.

Reply:
Updated.